web/firefoxcom.js

```diff
@@ -495,6 +495,72 @@ class MLManager {
   }
 }
 
+class SignatureStorage {
+  #signatures = null;
+
+  #handleSignature(data) {
+    return FirefoxCom.requestAsync("handleSignature", data);
+  }
+
+  async getAll() {
+    if (!this.#signatures) {
+      this.#signatures = new Map(
+        (await this.#handleSignature({ action: "get" }))?.map(
+          ({ uuid, description, signatureData }) => [
+            uuid,
+            { description, signatureData },
+          ]
+        ) || []
```

Comment:
Sorry, but this isn't a wonder of readability; maybe a temporary variable (or two) would help that?